src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/on_seek_media_clock_timer_notification_test.cc

```diff
@@ -0,0 +1,114 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
+
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/mock_application.h"
+#include "application_manager/mock_application_manager.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "application_manager/smart_object_keys.h"
+#include "gtest/gtest.h"
+#include "interfaces/MOBILE_API.h"
+#include "mobile/on_seek_media_clock_timer_notification.h"
+#include "smart_objects/smart_object.h"
+#include "utils/data_accessor.h"
+#include "utils/lock.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace on_seek_media_clock_timer_notification {
+
+using ::testing::_;
+using ::testing::Eq;
+using ::testing::Return;
+namespace am = ::application_manager;
+using am::commands::MessageSharedPtr;
+using sdl_rpc_plugin::commands::mobile::OnSeekMediaClockTimerNotification;
+
+using namespace am::commands;
+
+const uint32_t kConnectionKey = 2u;
+const uint32_t kCorrelationId = 2u;
+const uint32_t kAppID = 2u;
+const uint32_t kHours = 2u;
+const uint32_t kMinutes = 26u;
+const uint32_t kSeconds = 1u;
+
+typedef std::shared_ptr<OnSeekMediaClockTimerNotification> NotificationPtr;
+
+class OnSeekMediaClockTimerNotificationTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  OnSeekMediaClockTimerNotificationTest() {
+    CreateBasicMessage();
+    mock_app = CreateMockApp();
+  }
+
+ public:
+  MessageSharedPtr commands_msg;
+  MockAppPtr mock_app;
+
+ private:
+  void CreateBasicMessage() {
+    commands_msg = CreateMessage(smart_objects::SmartType_Map);
+    (*commands_msg)[am::strings::msg_params][am::strings::seekTime]
+                   [am::strings::hours] = kHours;
+    (*commands_msg)[am::strings::msg_params][am::strings::seekTime]
+                   [am::strings::minutes] = kMinutes;
+    (*commands_msg)[am::strings::msg_params][am::strings::seekTime]
+                   [am::strings::seconds] = kSeconds;
+    (*commands_msg)[am::strings::msg_params][am::strings::app_id] = kAppID;
+  }
+};
+
+TEST_F(OnSeekMediaClockTimerNotificationTest,
```

Comment:
If we want to cover code of this notification properly, we also need to add some negative cases to see difference. Particularly, no app case and seek not enabled case. Because without It it looks like Google mocks work well, and it's not enough, I believe.